dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/context/MultiThreadedContextServlet.java

```diff
@@ -0,0 +1,110 @@
+import jakarta.enterprise.concurrent.ManagedExecutorService;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.servlet.annotation.WebServlet;
+
+@SuppressWarnings("serial")
+@WebServlet("/context")
+@ApplicationScoped
+public class MultiThreadedContextServlet extends FATServlet {
+
+    @Resource
+    ManagedExecutorService managedExecutorService;
+
+    @Test
+    public void testContextTaskWrapping() throws InterruptedException, ExecutionException, TimeoutException {
+        //add some baggage to the current context
+        addBaggage("myKey1", "myValue1");
+
+        // execute my async sub-task
+        ExecutorService executorService = Context.taskWrapping(managedExecutorService);
+        Future<String> future = executorService.submit(new MyCallable());
+        String baggageValue = future.get(5, TimeUnit.SECONDS);
+        assertEquals("myValue1", baggageValue);
+
+        //check that the baggage was not modified by the sub-task
+        assertBaggage("myKey1", "myValue1");
+    }
+
+    @Test
+    public void testContextWrap() throws InterruptedException, ExecutionException, TimeoutException {
+        //add some baggage to the current context
+        addBaggage("myKey1", "myValue1");
+
+        // execute my async sub-task
+        Context context = Context.current();
+        ExecutorService executorService = context.wrap(managedExecutorService);
+        Future<String> future = executorService.submit(new MyCallable());
+        String baggageValue = future.get(5, TimeUnit.SECONDS);
+        assertEquals("myValue1", baggageValue);
+
+        //check that the baggage was not modified by the sub-task
+        assertBaggage("myKey1", "myValue1");
+    }
+
+    //check that the current context contains baggage with a certain value
+    private static void assertBaggage(String key, String expectedValue) {
+        Baggage baggage = Baggage.current();
+        String value = baggage.getEntryValue(key);
+        assertEquals(expectedValue, value);
+    }
+
+    //add some baggage to the current context
+    private static void addBaggage(String key, String value) {
+        BaggageBuilder builder = Baggage.builder();
+        builder.put(key, value);
+        Baggage baggage = builder.build();
+        baggage.makeCurrent();
+    }
+
+    private static class MyCallable implements Callable<String> {
+
+        /** {@inheritDoc} */
+        @Override
+        public String call() throws Exception {
+            //get the baggage and check the contents
+            Baggage baggage = Baggage.current();
+            String value = baggage.getEntryValue("myKey1");
+
+            //create a new baggage and make a new current context with it
+            //this new context should not affect the original one
+            addBaggage("myKey1", "myValue2");
+
+            //check that the new baggage is in the current context
+            assertBaggage("myKey1", "myValue2");
+
+            return value;
+        }
```

Comment:
Given that you now have an assert in this method anyway, it might make sense to assert `value` here as well, rather than returning it an asserting in the original method.